cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -210,25 +224,50 @@ runQueryTip (AnyConsensusModeParams cModeParams) network mOutFile = do
     CardanoMode -> do
       let localNodeConnInfo = LocalNodeConnectInfo cModeParams network sockPath
 
-      (chainTip, eLocalState) <- liftIO $
-        executeLocalStateQueryExprWithChainSync localNodeConnInfo Nothing $ \ntcVersion -> do
-          era <- queryExpr (QueryCurrentEra CardanoModeIsMultiEra)
-          eraHistory <- queryExpr (QueryEraHistory CardanoModeIsMultiEra)
-          mSystemStart <- if ntcVersion >= NodeToClientV_9
-            then Just <$> queryExpr QuerySystemStart
-            else return Nothing
-          return O.QueryTipLocalState
-            { O.era = era
-            , O.eraHistory = eraHistory
-            , O.mSystemStart = mSystemStart
-            }
+      eLocalState <- liftIO $ executeLocalStateQueryExpr localNodeConnInfo Nothing $ \ntcVersion -> do
+        era <- queryExpr (QueryCurrentEra CardanoModeIsMultiEra)
+        eraHistory <- queryExpr (QueryEraHistory CardanoModeIsMultiEra)
+        mHeaderStateTip <- if ntcVersion >= NodeToClientV_10
+          then Just <$> queryExpr QueryHeaderStateTip
+          else return Nothing
+        mSystemStart <- if ntcVersion >= NodeToClientV_9
+          then Just <$> queryExpr QuerySystemStart
+          else return Nothing
+        return O.QueryTipLocalState
+          { O.era = era
+          , O.eraHistory = eraHistory
+          , O.mSystemStart = mSystemStart
+          , O.mHeaderStateTip = mHeaderStateTip
+          }
 
       mLocalState <- hushM (first ShelleyQueryCmdAcquireFailure eLocalState) $ \e ->
         liftIO . T.hPutStrLn IO.stderr $ "Warning: Local state unavailable: " <> renderShelleyQueryCmdError e
 
+      chainTip <- case mLocalState >>= O.mHeaderStateTip of
+        Just headerStateTip ->
+          case headerStateTip of
+            Origin -> return Origin
+            At hst -> return $ At O.HeaderStateTipOutput
+              { O.slotNo = headerStateTipToSlotNo hst
+              , O.blockNo = headerStateTipToBlockNo hst
+              , O.headerHash = renderHeaderHash Proxy (headerStateTipToHeaderHash hst)
+              }
+        Nothing -> do
+          liftIO . T.hPutStrLn IO.stderr $
+            "Warning: Local header state query unavailable.  Falling back to chain sync query"
```

Comment:
Is the extra space intentional?

Reply:
An old habit I developed because my high school teacher insisted this is how sentences are supposed to be delimited 😆

Reply:
Hehe, yeah, I think that's the way typists break their sentences :) Even Emacs requires two spaces after the period to consider a sentence ended.